scripts/dts/gen_defines.py

```diff
@@ -298,20 +786,42 @@ def write_bus(node):
 def write_existence_flags(node):
     # Generate #defines of the form
     #
-    #   #define DT_INST_<instance no.>_<compatible string> 1
+    #   #define DT_EXISTS_<IDENT>      1
+    #
+    # For backwards compatibility, we also emit:
+    #
+    #   #define DT_<IDENT>             DT_EXISTS_<IDENT>
     #
     # for enabled nodes. These are flags for which devices exist.
-
-    for compat in node.compats:
-        instance_no = node.edt.compat2enabled[compat].index(node)
-        out(f"INST_{instance_no}_{str2ident(compat)}", 1)
-
-
-def node_ident(node):
-    # Returns an identifier for 'node'. Used e.g. when building macro names.
-
-    # TODO: Handle PWM on STM
-    # TODO: Better document the rules of how we generate things
+    # For example, for compat "vnd,device", this could look like:
+    #
+    #   #define DT_EXISTS_INST_0_VND_DEVICE 1
+
+    def write_existence_flags_for_ident(node, ident):
+        flag, legacy_flag = f"EXISTS_{ident}", ident
+        if flag not in existence_flags:
+            primary = out(flag, 1)
+            existence_flags.add(flag)
+        if legacy_flag not in existence_flags:
+            out_comment(f"Note: DT_{legacy_flag} is deprecated; use {primary}",
+                        blank_before=False)
+            out(legacy_flag, primary)
+            existence_flags.add(legacy_flag)
+
+    out_comment("Existence macros (these mark the node as enabled)",
+                blank_before=False)
+    for ident in node.z_inst_idents:
+        write_existence_flags_for_ident(node, ident)
+    for ident in node.z_label_idents:
```

Comment:
Would prefer if we did NOT generate `DT_NODELABEL_FOO 1` and only generated the `DT_EXISTS_NODELABEL_FOO 1`